pootle/apps/pootle_project/models.py

```diff
@@ -396,6 +395,10 @@ def clean(self):
                 _('"%s" cannot be used as a project code', self.code)
             )
 
+    def directory_exists(self):
```

Comment:
I'd like to agree because it seems simple and clear. But `Project` and `TranslationProject` keep logic how their paths are retrieved when `Directory` just have a path. 
Probably it can be done but not in this PR.